packages/core/src/deployment.ts

```diff
@@ -38,31 +49,42 @@ export async function resumeOrDeploy<T extends Deployment>(
 }
 
 export async function waitAndValidateDeployment(provider: EthereumProvider, deployment: Deployment): Promise<void> {
+  const { txHash, address } = deployment;
   const startTime = Date.now();
 
-  // Poll for 60 seconds with a 5 second poll interval.
-  // TODO: Make these parameters configurable.
-  while (Date.now() - startTime < 60e3) {
-    debug('verifying deployment tx mined', deployment.txHash);
-    const tx = await getTransactionByHash(provider, deployment.txHash);
-    if (tx === null) {
-      throw new InvalidDeployment(deployment);
-    }
-    if (tx.blockHash !== null) {
-      debug('verifying deployment tx succeeded', deployment.txHash);
-      const code = await getCode(provider, deployment.address);
-      if (code === '0x') {
+  if (txHash !== undefined) {
+    // Poll for 60 seconds with a 5 second poll interval.
+    // TODO: Make these parameters configurable.
+    const pollTimeout = 60e3;
+    const pollInterval = 5e3;
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const elapsedTime = Date.now() - startTime;
+      if (elapsedTime >= pollTimeout) {
+        // A timeout is NOT an InvalidDeployment
+        throw new TransactionMinedTimeout(deployment);
+      }
+      debug('verifying deployment tx mined', txHash);
+      const tx = await getTransactionByHash(provider, txHash);
+      if (tx === null) {
         throw new InvalidDeployment(deployment);
-      } else {
-        return;
       }
+      if (tx.blockHash !== null) {
+        debug('verifying deployment tx mined succeeded', txHash);
+        break;
+      }
+      debug('waiting for deployment tx mined', txHash);
+      await sleep(pollInterval);
     }
-    debug('waiting for deployment tx mined', deployment.txHash);
-    await sleep(5e3);
   }
 
-  // A timeout is NOT an InvalidDeployment
-  throw new TransactionMinedTimeout(deployment);
+  if (await hasCode(provider, address)) {
+    debug('verifying deployment succeeded', txHash);
```

Comment:
This debug log was before the call to `getCode`, so it should go before `hasCode`. Also, the same comment I had above about the placement of "succeeded".

```suggestion
  debug('succeeded verifying deployment', txHash);
  if (await hasCode(provider, address)) {
```